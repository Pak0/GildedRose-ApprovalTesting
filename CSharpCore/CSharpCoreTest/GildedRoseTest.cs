using Xunit;
using System.Collections.Generic;
using FluentAssertions;
using ApprovalTests.Reporters;
using ApprovalTests;
using Newtonsoft.Json;
using ApprovalTests.Combinations;

namespace CSharpCore.Test
{
    public class GildedRoseTest
    {
        [Fact]
        [UseReporter(typeof(VisualStudioReporter))]
        public void ShouldHaveItemWithExpectedName()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo2", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();

            Approvals.Verify(Items[0].ToString());
        }

        [Fact]
        [UseReporter(typeof(VisualStudioReporter))]
        public void ShouldHaveItemWithExpectedName_JsonConvert()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();

            Approvals.Verify(JsonConvert.SerializeObject(Items[0]));
        }

        [Fact]
        [UseReporter(typeof(VisualStudioReporter))]
        public void ShouldHaveItemWithExpectedName_VerifyJson()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "Sulfuras, Hand of Ragnaros", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();

            Approvals.VerifyJson(JsonConvert.SerializeObject(Items[0]));
        }

        [Fact]
        [UseReporter(typeof(VisualStudioReporter))]
        public void ShouldHaveItemWithAllCombinations()
        {
            IList<string> Items = new List<string> { "Aged Brie", 
                                                     "Sulfuras, Hand of Ragnaros", 
                                                     "Backstage passes to a TAFKAL80ETC concert", 
                                                     "foo" };
            IList<int> SellIns = new List<int> { -1, 0, 5, 6, 10, 11 };
            IList<int> Qualities = new List<int> { 0, 49, 50, 51 };

            CombinationApprovals.VerifyAllCombinations(doItemCheck, Items, SellIns, Qualities);
        }

        private string doItemCheck(string name, int sellIn, int quality)
        {
            IList<Item> Items = new List<Item> { new Item { Name = name, SellIn = sellIn, Quality = quality } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();

            return JsonConvert.SerializeObject(Items[0]);
        }
    }
}