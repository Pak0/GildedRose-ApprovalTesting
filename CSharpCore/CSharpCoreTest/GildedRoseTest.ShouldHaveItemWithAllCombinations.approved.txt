[Aged Brie,-1,0] => {"Name":"Aged Brie","SellIn":-2,"Quality":2}
[Aged Brie,-1,49] => {"Name":"Aged Brie","SellIn":-2,"Quality":50}
[Aged Brie,-1,50] => {"Name":"Aged Brie","SellIn":-2,"Quality":50}
[Aged Brie,-1,51] => {"Name":"Aged Brie","SellIn":-2,"Quality":51}
[Aged Brie,0,0] => {"Name":"Aged Brie","SellIn":-1,"Quality":2}
[Aged Brie,0,49] => {"Name":"Aged Brie","SellIn":-1,"Quality":50}
[Aged Brie,0,50] => {"Name":"Aged Brie","SellIn":-1,"Quality":50}
[Aged Brie,0,51] => {"Name":"Aged Brie","SellIn":-1,"Quality":51}
[Aged Brie,5,0] => {"Name":"Aged Brie","SellIn":4,"Quality":1}
[Aged Brie,5,49] => {"Name":"Aged Brie","SellIn":4,"Quality":50}
[Aged Brie,5,50] => {"Name":"Aged Brie","SellIn":4,"Quality":50}
[Aged Brie,5,51] => {"Name":"Aged Brie","SellIn":4,"Quality":51}
[Aged Brie,6,0] => {"Name":"Aged Brie","SellIn":5,"Quality":1}
[Aged Brie,6,49] => {"Name":"Aged Brie","SellIn":5,"Quality":50}
[Aged Brie,6,50] => {"Name":"Aged Brie","SellIn":5,"Quality":50}
[Aged Brie,6,51] => {"Name":"Aged Brie","SellIn":5,"Quality":51}
[Aged Brie,10,0] => {"Name":"Aged Brie","SellIn":9,"Quality":1}
[Aged Brie,10,49] => {"Name":"Aged Brie","SellIn":9,"Quality":50}
[Aged Brie,10,50] => {"Name":"Aged Brie","SellIn":9,"Quality":50}
[Aged Brie,10,51] => {"Name":"Aged Brie","SellIn":9,"Quality":51}
[Aged Brie,11,0] => {"Name":"Aged Brie","SellIn":10,"Quality":1}
[Aged Brie,11,49] => {"Name":"Aged Brie","SellIn":10,"Quality":50}
[Aged Brie,11,50] => {"Name":"Aged Brie","SellIn":10,"Quality":50}
[Aged Brie,11,51] => {"Name":"Aged Brie","SellIn":10,"Quality":51}
[Sulfuras, Hand of Ragnaros,-1,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":-1,"Quality":0}
[Sulfuras, Hand of Ragnaros,-1,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":-1,"Quality":49}
[Sulfuras, Hand of Ragnaros,-1,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":-1,"Quality":50}
[Sulfuras, Hand of Ragnaros,-1,51] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":-1,"Quality":51}
[Sulfuras, Hand of Ragnaros,0,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":0,"Quality":0}
[Sulfuras, Hand of Ragnaros,0,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":0,"Quality":49}
[Sulfuras, Hand of Ragnaros,0,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":0,"Quality":50}
[Sulfuras, Hand of Ragnaros,0,51] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":0,"Quality":51}
[Sulfuras, Hand of Ragnaros,5,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":5,"Quality":0}
[Sulfuras, Hand of Ragnaros,5,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":5,"Quality":49}
[Sulfuras, Hand of Ragnaros,5,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":5,"Quality":50}
[Sulfuras, Hand of Ragnaros,5,51] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":5,"Quality":51}
[Sulfuras, Hand of Ragnaros,6,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":6,"Quality":0}
[Sulfuras, Hand of Ragnaros,6,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":6,"Quality":49}
[Sulfuras, Hand of Ragnaros,6,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":6,"Quality":50}
[Sulfuras, Hand of Ragnaros,6,51] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":6,"Quality":51}
[Sulfuras, Hand of Ragnaros,10,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":10,"Quality":0}
[Sulfuras, Hand of Ragnaros,10,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":10,"Quality":49}
[Sulfuras, Hand of Ragnaros,10,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":10,"Quality":50}
[Sulfuras, Hand of Ragnaros,10,51] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":10,"Quality":51}
[Sulfuras, Hand of Ragnaros,11,0] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":11,"Quality":0}
[Sulfuras, Hand of Ragnaros,11,49] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":11,"Quality":49}
[Sulfuras, Hand of Ragnaros,11,50] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":11,"Quality":50}
[Sulfuras, Hand of Ragnaros,11,51] => {"Name":"Sulfuras, Hand of Ragnaros","SellIn":11,"Quality":51}
[Backstage passes to a TAFKAL80ETC concert,-1,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-2,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,-1,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-2,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,-1,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-2,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,-1,51] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-2,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,0,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-1,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,0,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-1,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,0,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-1,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,0,51] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":-1,"Quality":0}
[Backstage passes to a TAFKAL80ETC concert,5,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":4,"Quality":3}
[Backstage passes to a TAFKAL80ETC concert,5,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":4,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,5,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":4,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,5,51] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":4,"Quality":51}
[Backstage passes to a TAFKAL80ETC concert,6,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":5,"Quality":2}
[Backstage passes to a TAFKAL80ETC concert,6,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":5,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,6,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":5,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,6,51] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":5,"Quality":51}
[Backstage passes to a TAFKAL80ETC concert,10,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":9,"Quality":2}
[Backstage passes to a TAFKAL80ETC concert,10,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":9,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,10,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":9,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,10,51] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":9,"Quality":51}
[Backstage passes to a TAFKAL80ETC concert,11,0] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":10,"Quality":1}
[Backstage passes to a TAFKAL80ETC concert,11,49] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":10,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,11,50] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":10,"Quality":50}
[Backstage passes to a TAFKAL80ETC concert,11,51] => {"Name":"Backstage passes to a TAFKAL80ETC concert","SellIn":10,"Quality":51}
[foo,-1,0] => {"Name":"foo","SellIn":-2,"Quality":0}
[foo,-1,49] => {"Name":"foo","SellIn":-2,"Quality":47}
[foo,-1,50] => {"Name":"foo","SellIn":-2,"Quality":48}
[foo,-1,51] => {"Name":"foo","SellIn":-2,"Quality":49}
[foo,0,0] => {"Name":"foo","SellIn":-1,"Quality":0}
[foo,0,49] => {"Name":"foo","SellIn":-1,"Quality":47}
[foo,0,50] => {"Name":"foo","SellIn":-1,"Quality":48}
[foo,0,51] => {"Name":"foo","SellIn":-1,"Quality":49}
[foo,5,0] => {"Name":"foo","SellIn":4,"Quality":0}
[foo,5,49] => {"Name":"foo","SellIn":4,"Quality":48}
[foo,5,50] => {"Name":"foo","SellIn":4,"Quality":49}
[foo,5,51] => {"Name":"foo","SellIn":4,"Quality":50}
[foo,6,0] => {"Name":"foo","SellIn":5,"Quality":0}
[foo,6,49] => {"Name":"foo","SellIn":5,"Quality":48}
[foo,6,50] => {"Name":"foo","SellIn":5,"Quality":49}
[foo,6,51] => {"Name":"foo","SellIn":5,"Quality":50}
[foo,10,0] => {"Name":"foo","SellIn":9,"Quality":0}
[foo,10,49] => {"Name":"foo","SellIn":9,"Quality":48}
[foo,10,50] => {"Name":"foo","SellIn":9,"Quality":49}
[foo,10,51] => {"Name":"foo","SellIn":9,"Quality":50}
[foo,11,0] => {"Name":"foo","SellIn":10,"Quality":0}
[foo,11,49] => {"Name":"foo","SellIn":10,"Quality":48}
[foo,11,50] => {"Name":"foo","SellIn":10,"Quality":49}
[foo,11,51] => {"Name":"foo","SellIn":10,"Quality":50}
